sdk/cosmos/azure-cosmos/azure/cosmos/auth.py

```diff
@@ -140,9 +141,12 @@ def __GetAuthorizationTokenUsingResourceTokens(resource_tokens, path, resource_i
             "conflicts",
             "offers",
         ]
+
         # Get the last resource id or resource name from the path and get it's token from resource_tokens
-        for one_part in reversed(path_parts):
-            if not one_part in resource_types and one_part in resource_tokens:
-                return resource_tokens[one_part]
+        for i in range(len(path_parts), 1, -1):
+            segment = path_parts[i-1]
```

Comment:
segment will never be path_parts[0] - this is the expected behaviour?

Reply:
yes, we expect the path to be in "dbs/.../colls/..." format. The "dbs/" segment shouldn't be considered.